### ISSUE TITLE
Bug: Out of Gas error on calling view function which uses msg.sender()

### ISSUE THREAD

**Issue by blockchainbuddha** [blockchainbuddha/one-click-dApp#33]
Example: myEarnings() on 
http://oneclickdapp.com/camel-join/

Error message `Unhandled promise rejection Error: "Returned values aren't valid, did it run Out of Gas?"`

**Comment by blockchainbuddha** [comment]
If you deploy the same contract on local ganache-cli, then there is no error!

**Comment by blockchainbuddha** [comment]
Even weirder is it works on Ropsten, but not on Mainnet
ropsten: http://oneclickdapp.com/vampire-common/
mainnet: http://oneclickdapp.com/bison-career/

**Comment by blockchainbuddha** [comment]
This seems to be an issue with the ABI not having a name for the output for a view function. I fixed this by giving the output a name "earnings" in the the ABI, and creating a new dApp with this ABI.

All, please add a name to your outputs in your ABI, until a fix is implemented to add it automatically upon creation
```
	{
		"constant": true,
		"inputs": [],
		"name": "myEarnings",
		"outputs": [
			{
				**"name": "earnings",**
				"type": "uint256"
			}
		],
		"payable": false,
		"stateMutability": "view",
		"type": "function"
	},
	{
		"inputs": [],
		"payable": false,
		"stateMutability": "nonpayable",
		"type": "constructor"
	}
```

**Comment by blockchainbuddha** [comment]
This has been fixed in commit 5061a41
Now, all unnamed outputs in all view functions are automatically given a name.
